pycrires/pipeline.py

```diff
@@ -3884,6 +3884,140 @@ def correct_wavelengths(
                 plt.tight_layout()
                 plt.savefig(f'{output_dir}/correlation_map_{out_label}.png')
 
+    @typechecked
+    def util_extract_2d(self, nod_ab: str = 'A', 
+                        verbose: bool = True,
+                        extraction_length = None, 
+                        spatial_oversampling = 1.,
+                        util_extract_oversample = 12
+                        ) -> None:
+        """
+        Method for extracting spectra from the products of obs_nodding, while retaining
+        the spatial information.
+        Make sure that you have run obs_nodding first. 
+        The spectra are extracted by running util_extract over a range of slit fractions.
+
+        Parameters
+        ----------
+        nod_ab : str
+            Nod position with the spectra of which the wavelength
+            solution will be corrected.
+        verbose : bool
+            Print output produced by ``esorex``.
+        extraction_length: float
+            Spatial extent over which to extract the spectrum in arcseconds.
+            If 'None' then the pixel scale of the instrument is used.
+        spatial_oversampling: float
+            Oversampling factor for the extraction along the slit length.
+            Example: If spatial_oversampling=2 is being used the end result
+            will have twice as many spatial pixels as the original data.
+        util_extract_oversample: int
+            Oversampling to use for util_extract.
+        Returns
+        -------
+        NoneType
+            None
+        """
+
+        self._print_section("Process nodding frames", recipe_name="cr2res_util_extract_2d")
+
+        fits_files = sorted(list(self.file_dict[f'OBS_NODDING_COMBINED{nod_ab}'].keys()))
+
+        # Iterate over nod A exposures
+        for count_exp,fits_file in enumerate(fits_files):
+            
+            # Prepare SOF file
+            sof_file = pathlib.Path(self.product_folder / f"files_{count_exp:03d}.sof")
```

Comment:
I have changed the output folder of `obs_nodding` to "product/obs_nodding". I think that we can do the same for `util_extract_2d`, so change the output folder to "product/util_extract_2d".

Reply:
Done